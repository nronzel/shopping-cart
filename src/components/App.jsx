import React from "react";
import { BrowserRouter } from "react-router-dom";
import RouteSwitch from "./RouteSwitch";

const App = () => {
    return (
        <BrowserRouter>
            <RouteSwitch />
        </BrowserRouter>
    );
};

export default App;
