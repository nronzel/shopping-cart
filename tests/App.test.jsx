import { render, screen } from '@testing-library/react'
import App from '../src/components/App.jsx'
import React from "react";
import { ChakraProvider } from '@chakra-ui/react';

describe("App test", () => {
    it("Render the app", () => {
        const { container } = render(<ChakraProvider><App /></ChakraProvider>);

        expect(container).toMatchSnapshot();

    })
})
