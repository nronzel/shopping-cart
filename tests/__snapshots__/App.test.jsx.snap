// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App test > Render the app 1`] = `
<div>
  <div
    class="css-9dbaa4"
  >
    <div
      class="css-1lo8p5r"
    >
      <div
        class="css-2pts9x"
      >
        <div
          class="css-4d8kqp"
        >
          <h2
            class="chakra-heading css-1tqj5lc"
          >
            <a
              href="/"
            >
              keebstore
            </a>
          </h2>
          <p
            class="chakra-text css-gafbmw"
          >
            <a
              href="/products"
            >
              shop
            </a>
          </p>
          <div
            class="chakra-stack css-1fkrl9x"
          >
            <a
              class="chakra-link css-spn4bz"
              href="/cart"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-cart-shopping fa-lg "
                data-icon="cart-shopping"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 24C0 10.7 10.7 0 24 0H69.5c22 0 41.5 12.8 50.6 32h411c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3H170.7l5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5H488c13.3 0 24 10.7 24 24s-10.7 24-24 24H199.7c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5H24C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <button
              class="chakra-button css-i857na"
              type="button"
            >
              <svg
                class="chakra-icon css-in1cf3"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21.4,13.7C20.6,13.9,19.8,14,19,14c-5,0-9-4-9-9c0-0.8,0.1-1.6,0.3-2.4c0.1-0.3,0-0.7-0.3-1 c-0.3-0.3-0.6-0.4-1-0.3C4.3,2.7,1,7.1,1,12c0,6.1,4.9,11,11,11c4.9,0,9.3-3.3,10.6-8.1c0.1-0.3,0-0.7-0.3-1 C22.1,13.7,21.7,13.6,21.4,13.7z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-j7qwjs"
    >
      <div
        class="css-g96nnt"
        draggable="false"
      >
        <img
          alt="keyboard"
          class="chakra-image css-111xjgq"
          draggable="false"
          src="/bg.jpg"
        />
        <div
          class="css-telqaq"
        >
          <h2
            class="chakra-heading css-1q3nbl9"
          >
            sale ends soon!
          </h2>
          <p
            class="chakra-text css-3zrjl2"
          >
            get the hottest keyboards and accessories before they are gone.
          </p>
          <a
            class="chakra-link css-bbwh0s"
            href="/products"
          >
            <button
              class="chakra-button css-ir6p0w"
              type="button"
            >
              shop now
            </button>
          </a>
        </div>
      </div>
      <div
        class="css-1dbyw6p"
      >
        <div
          class="css-1k5rtfp"
          draggable="false"
        >
          <a
            class="chakra-link css-spn4bz"
            href="/products"
          >
            <img
              alt="/categories/switches.jpg"
              class="chakra-image css-1dnjb3l"
              draggable="false"
              src="/categories/switches.jpg"
            />
          </a>
        </div>
        <div
          class="css-1k5rtfp"
          draggable="false"
        >
          <a
            class="chakra-link css-spn4bz"
            href="/products"
          >
            <img
              alt="/categories/chasis.jpg"
              class="chakra-image css-1dnjb3l"
              draggable="false"
              src="/categories/chasis.jpg"
            />
          </a>
        </div>
        <div
          class="css-1k5rtfp"
          draggable="false"
        >
          <a
            class="chakra-link css-spn4bz"
            href="/products"
          >
            <img
              alt="/categories/keycaps.jpg"
              class="chakra-image css-1dnjb3l"
              draggable="false"
              src="/categories/keycaps.jpg"
            />
          </a>
        </div>
        <div
          class="css-1k5rtfp"
          draggable="false"
        >
          <a
            class="chakra-link css-spn4bz"
            href="/products"
          >
            <img
              alt="/categories/cables.jpg"
              class="chakra-image css-1dnjb3l"
              draggable="false"
              src="/categories/cables.jpg"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="css-17xejub"
    />
    <div
      class="css-p9mjwf"
    >
      <p
        class="chakra-text css-119zpey"
      >
        nronzel Â© 2023 All Rights Reserved
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
